packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.js

```diff
@@ -2,52 +2,86 @@ import React from 'react';
 import { shape, string, bool, func } from 'prop-types';
 import { RadioGroup } from 'informed';
 
+import { usePaymentMethods } from '@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/usePaymentMethods';
+
+import { mergeClasses } from '../../../classify';
 import Radio from '../../RadioGroup/radio';
 import CreditCard from './creditCard';
-import { mergeClasses } from '../../../classify';
-
+import paymentMethodOperations from './paymentMethods.gql';
 import defaultClasses from './paymentMethods.css';
 
+const PAYMENT_METHOD_COMPONENTS_BY_CODE = {
+    braintree: CreditCard
+    // checkmo: CheckMo,
+    // etc
+};
+
 const PaymentMethods = props => {
     const {
         classes: propClasses,
         reviewOrderButtonClicked,
-        selectedPaymentMethod,
+        setDoneEditing,
         onPaymentSuccess,
         onPaymentError,
         resetReviewOrderButtonClicked
     } = props;
 
     const classes = mergeClasses(defaultClasses, propClasses);
 
-    const creditCard =
-        selectedPaymentMethod === 'braintree' ? (
-            <CreditCard
-                shouldSubmit={reviewOrderButtonClicked}
-                brainTreeDropinContainerId={
-                    'checkout-page-braintree-dropin-container'
-                }
-                onPaymentSuccess={onPaymentSuccess}
-                onPaymentError={onPaymentError}
-                resetShouldSubmit={resetReviewOrderButtonClicked}
-            />
-        ) : null;
+    const talonProps = usePaymentMethods({
+        ...paymentMethodOperations
+    });
+
+    const {
+        availablePaymentMethods,
+        currentSelectedPaymentMethod,
+        initialSelectedMethod,
+        isLoading
+    } = talonProps;
+
+    if (isLoading) {
+        return null;
+    }
+
+    const radios = availablePaymentMethods.map(({ code, title }) => {
+        // If we don't have an implementation for a method type, ignore it.
+        if (!Object.keys(PAYMENT_METHOD_COMPONENTS_BY_CODE).includes(code)) {
+            return;
+        }
+
+        const isSelected = currentSelectedPaymentMethod === code;
+        const Component = PAYMENT_METHOD_COMPONENTS_BY_CODE[code] || null;
+
+        return (
+            <div key={code} className={classes.payment_method}>
+                <Radio
+                    label={title}
+                    value={code}
+                    classes={{
+                        label: classes.radio_label
+                    }}
+                    checked={isSelected}
+                />
+                {isSelected && (
```

Comment:
Try to avoid multiline expressions in JSX like this.

Reply:
I'll change this. I wrote it as is because of the naming of the generic `Component` variable above. I wasn't really sure what to call it but I'll figure something out.